huaweicloud/services/acceptance/cph/resource_huaweicloud_cph_server_test.go

```diff
@@ -0,0 +1,202 @@
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving CphServer: %s", err)
+	}
+
+	getCphServerRespBody, err := utils.FlattenResponse(getCphServerResp)
+	if err != nil {
+		return nil, err
+	}
+	statusRaw := utils.PathSearch("status", getCphServerRespBody, nil)
+
+	if fmt.Sprint(statusRaw) == "6" {
+		return nil, golangsdk.ErrDefault404{}
+	}
+
+	return getCphServerRespBody, nil
+}
+
+func TestAccCphServer_basic(t *testing.T) {
+	var obj interface{}
+
+	name := acceptance.RandomAccResourceName()
+	rName := "huaweicloud_cph_server.test"
+
+	rc := acceptance.InitResourceCheck(
+		rName,
+		&obj,
+		getCphServerResourceFunc,
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		CheckDestroy:      rc.CheckResourceDestroy(),
+		Steps: []resource.TestStep{
+			{
+				Config: testCphServer_basic(name),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+					resource.TestCheckResourceAttrPair(rName, "vpc_id", "huaweicloud_vpc.test", "id"),
+					resource.TestCheckResourceAttrPair(rName, "subnet_id", "huaweicloud_vpc_subnet.test", "id"),
+					resource.TestCheckResourceAttrPair(rName, "vpc_id", "huaweicloud_vpc.test", "id"),
+					resource.TestCheckResourceAttrSet(rName, "availability_zone"),
+					resource.TestCheckResourceAttrSet(rName, "order_id"),
+					resource.TestCheckResourceAttrSet(rName, "addresses.#"),
+					resource.TestCheckResourceAttrSet(rName, "status"),
+				),
+			},
+			{
+				Config: testCphServer_basic_update(name, name+"update"),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+					resource.TestCheckResourceAttr(rName, "name", name+"update"),
+				),
+			},
+			{
+				ResourceName:            rName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"image_id", "public_ip", "auto_renew", "period", "period_unit"},
+			},
+		},
+	})
+}
+
+func testCphServerBase(name string) string {
+	return fmt.Sprintf(`
+resource "huaweicloud_vpc" "test" {
+  name = "%s"
+  cidr = "192.168.0.0/16"
+}
+
+resource "huaweicloud_vpc_subnet" "test" {
+  name       = "%s"
+  cidr       = "192.168.0.0/24"
+  vpc_id     = huaweicloud_vpc.test.id
+  gateway_ip = "192.168.0.1"
+}
+
+resource "huaweicloud_kps_keypair" "test" {
+  name        = "%s"
+  description = "%s"
+}
+`, name, name, name, name)
+}
+
+func testCphServer_basic(name string) string {
+	return fmt.Sprintf(`
+%s
+
+resource "huaweicloud_cph_server" "test" {
+  name          = "%s"
+  server_flavor = "physical.rx1.xlarge"
+  phone_flavor  = "rx1.cp.c15.d46.e1v1"
+  image_id      = "22110120221209e101210a3000000b02"
+  keypair_name  = huaweicloud_kps_keypair.test.name
+
+  vpc_id    = huaweicloud_vpc.test.id
+  subnet_id = huaweicloud_vpc_subnet.test.id
+
+  public_ip {
```

Comment:
should not run with public_ip as we can't control the security group, this will cause security issue.

Reply:
have modified the default security group